project/src/main/java/com/google/autograder/servlets/helpers/API.java

```diff
@@ -68,6 +85,33 @@ public static String getCurrentUserAPIAuthorization() {
         return authorization;
     }
 
+    public static Optional<HttpURLConnection> getAuthenticatedRequest(String method, String endpoint) {
+        String authorization = getCurrentUserAPIAuthorization();
+        HttpURLConnection connection = null;
```

Comment:
Leave `connection` uninitialized since the only way to initialize it is to open a valid connection. Otherwise if you accidentally remove the return statement on line 100 it becomes a compile-time error to reference connection rather than potentially referencing a null connection. Variables should also be declared as late as possible, in this case right before the try block on line 96

Reply:
+1 Okay that makes sense, especially with a connection object not being valid if the connection is not even opened.